src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -36,6 +36,11 @@ internal class WorkloadInstallerFactory
                 return new NetSdkMsiInstaller();
             }
 
+            if (!CanWriteToDotnetRoot(dotnetDir))
```

Comment:
Don't know about other platforms, but for Windows, UnauthorizedException is typically used since that also sets up the appropriate E_ACCESS_DENIED HResult.